app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -4110,19 +4134,85 @@ public AppCompatActivity getParentActivity() {
         return (AppCompatActivity) ((ViewGroup) binding.getRoot().getParent()).getContext();
     }
 
-    private void useVideoSource(final boolean video) {
-        if (playQueue == null || isAudioOnly == !video || audioPlayerSelected()) {
+    private void useVideoSource(final boolean videoEnabled) {
```

Comment:
As pointed out by SonarQube the code here is very complex:
https://sonarcloud.io/project/issues?id=TeamNewPipe_NewPipe&open=AX5de2YtRtLRntykaKph&pullRequest=7349&resolved=false&types=CODE_SMELL

It would be great if we could it make simpler / break it down. I did a proposal down below.